src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -118,7 +118,7 @@ public function validate($value, Constraint $constraint)
      */
     private function checkMX($host)
     {
-        return checkdnsrr($host, 'MX');
+        return ('' !== $host) && checkdnsrr($host, 'MX');
```

Comment:
`empty` is just too broad. For instance, it includes the `0` string as being `false`. Sometimes, that not desirable. So, we prefer to be very explicit about our expectations.